web3/contracts/FincubeDAO.sol

```diff
@@ -0,0 +1,383 @@
+     * @param status Whether the member is approved or not.
+     */
+    struct Member {
+        string memberURI;
+        bool status;
+    }
+    /**
+     * @dev Represents the different types of proposals that can be created.
+     */
+    enum ProposalType {
+        NewMemberProposal,
+        TokenAddressProposal,
+        RoyaltyProposal
+    }
+
+    /**
+     * @dev Represents a proposal within the DAO.
+     * @param proposer The address of the member who created the proposal.
+     * @param voteStart The timestamp when the voting period starts.
+     * @param voteDuration The duration of the voting period in seconds.
+     * @param executed Whether the proposal has been executed or not.
+     * @param canceled Whether the proposal has been canceled or not.
+     * @param data The data associated with the proposal (e.g., new member address, token address).
+     * @param yesvotes The number of "yes" votes for the proposal.
+     * @param novotes The number of "no" votes for the proposal.
+     */
+    struct Proposal {
+        address proposer;
+        uint48 voteStart;
+        uint48 voteDuration;
+        bool executed;
+        bool canceled;
+        bytes data;
+        uint256 yesvotes;
+        uint256 novotes;
+    }
+
+    /**
+     * @dev Stores the vote information for a proposal.
+     * @param isYesVote Mapping of addresses to whether they voted "yes" or not.
+     * @param isNoVote Mapping of addresses to whether they voted "no" or not.
+     */
+    struct ProposalVotes {
+        mapping(address => bool) isYesVote;
+        mapping(address => bool) isNoVote;
+    }
+    mapping(address => Member) private members;
+    mapping(uint256 => Proposal) private proposals;
+    mapping(uint256 => ProposalVotes) private proposalVotes;
+    mapping(uint256 => ProposalType) private proposalType;
+    address public tokenAddress;
+
+    /**
+     * @notice Initializes the contract with the owner as the first member.
+     * @param _ownerURI The URI that identifies the owner member.
+     */
+    constructor(string memory _ownerURI) Ownable(msg.sender) {
+        Member memory member;
+        member.memberURI = _ownerURI;
+        member.status = true;
+        members[msg.sender] = member;
+        memberCount = 1;
+    }
+
+    /**
+     * @notice Returns the voting delay period in seconds.
+     * @dev This function is marked as `pure` because it does not read or modify the contract's state.
+     * @return The voting delay period in seconds (currently 5 seconds).
+     */
+    function votingDelay() public pure returns (uint256) {
+        return 5; // 5 seconds
+    }
+
+    /**
+     * @notice Returns the voting period duration in seconds.
+     * @dev This function is marked as `pure` because it does not read or modify the contract's state.
+     * @return The voting period duration in seconds (currently 60 seconds).
+     */
+    function votingPeriod() public pure returns (uint256) {
```

Comment:
I think it should be internal. 
@SampadSikder / @mashiat0808    ?

Reply:
This will be public as the entities participating in voting must know when the voting will start and the duration of voting